design/cstor/v1/cstorpool.md

```diff
@@ -0,0 +1,504 @@
+# Moving CSPC and CSPI Schema to cstor group and version v1 from openebs group and version v1alpha1
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [Current State of Things](#current-state-of-things)
+* [CSPC and CSPI Schema Proposal](#cspc-schema-proposal)
+* [CRD Migration](#crd-migration)
+
+## Introduction
+
+This proposal highlights the limitations of current CSPC and CSPI schema and
+proposes improvements for the same.
+Please refer to the following link for cstor operator design document. 
+https://github.com/openebs/openebs/blob/master/contribute/design/1.x/cstor-operator/doc.md
+
+# Goals
+
+The major goal of this document is to freeze the schema for CSPC and CSPI schema.
```

Comment:
Here is suggested content for the Goals section:

OpenEBS cStor Data Engine has been in production for over a year now, with multiple cStor custom resource schema revisions. The goal of this design is to incorporate all the feedback received on the alpha and beta versions of cStor custom resources and define a version 1 schema.  

The schema proposed in this document encompasses the following requirements:
- Include the best practicies followed by users with SPC into the schema itself. 
- Rename the SPC and CSP to CSPC and CSPI based on the feedback received. 
- Enhance the elements of CSPC and CSPI to enable performing Day 2 operations declaratively: 
  * Scaling up/down of cStor Pools 
  * Expanding the capacity of cStor Pools 
  * Replacing the failed block devices in cStor Pools. 
  * Add supportability features via Status Conditions and Events 
  * Capture the history / log of upgrades